src/guiguts/misc_dialogs.py

```diff
@@ -431,6 +445,184 @@ def add_orphan_commands(cls) -> None:
         )
         menubar_metadata().add_checkbutton_orphan("Tooltips", PrefKey.SHOW_TOOLTIPS)
 
+    def create_color_rows(self) -> None:
+        """Create row of widgets for each configurable color.
+        If widgets already exist, delete them first.
+        """
+        sample_tag_name = "sample_tag"
+        dark_theme = themed_style().is_dark_theme()
+
+        for widget in self.colors_frame.winfo_children():
+            widget.destroy()
+
+        for row, key in enumerate(self.color_settings):
+            style_dict = (
+                self.color_settings[key].dark
+                if dark_theme
+                else self.color_settings[key].light
+            )
+            default_dict = (
+                self.default_colors[key].dark
+                if dark_theme
+                else self.default_colors[key].light
+            )
+
+            # Description sample as a Text widget
+            desc = tk.Text(
+                self.colors_frame,
+                background=maintext()["background"],
+                font=maintext()["font"],
+                width=20,
+                height=1,
+                relief="flat",
+                bd=0,
+                exportselection=False,
+                highlightthickness=0,
+                padx=0,
+                pady=0,
+            )
+            desc.grid(row=row, column=0, sticky="NSEW", padx=2, pady=2)
+            desc.insert("1.0", self.color_settings[key].description)
+            desc.tag_configure(sample_tag_name, style_dict)
+            desc.tag_add(sample_tag_name, "1.0", "1.end")
+            desc.config(state="disabled")
+            ToolTip(
+                desc,
+                "Left-click to set text color. Right-click or Shift-left-click to set background",
+            )
+
+            # Colorpicker bindings
+            def click_callback(
+                attr: str,
+                key: ColorKey = key,
+                widget: tk.Text = desc,
+                style_dict: StyleDict = style_dict,
+            ) -> None:
+                current = str(style_dict.get(attr, "#000000"))
+                color = colorchooser.askcolor(color=current, parent=self)[1]
+                if color:
+                    style_dict[attr] = color
+                    widget.config(state="normal")
+                    widget.tag_configure(sample_tag_name, style_dict)
+                    widget.config(state="disabled")
+                    self.update_color(key)
+
+            desc.bind(
+                "<Button-1>",
+                lambda _, cb=click_callback: cb("foreground"),  # type:ignore[misc]
+            )
+            desc.bind(
+                "<Button-3>",
+                lambda _, cb=click_callback: cb("background"),  # type:ignore[misc]
+            )
+            desc.bind(
```

Comment:
As discussed earlier, right-click on macOS doesn't work. Adding `<Button-2>` binding would fix that. If `<Shift-Button-1>` was only added to address that shortcoming, then potentially it could be removed.

Reply:
Thanks - we do add the button 2 binding when the code attempts a button 3 binding in some places, but I forgot to include it in this case. I'll add it.